### ISSUE TITLE
[REQUEST] Remove calls to debug intrinsics from LLVM IR output

### ISSUE THREAD

**Issue by JohanEngelen** [compiler-explorer/compiler-explorer#1963]
Clang and LDC provide LLVM IR output, which is very convenient in compiler explorer, but the result contains needless (to the user) calls to debug intrinsic `llvm.dbg.declare`:
```llvm
define hidden i32 @_Z3fooPi(i32* %0) #0 !dbg !7 {
  %2 = alloca i32*, align 4
  store i32* %0, i32** %2, align 4
  call void @llvm.dbg.declare(metadata i32** %2, metadata !13, metadata !DIExpression()), !dbg !14 ; <<-----
  %3 = load i32*, i32** %2, align 4, !dbg !15
  %4 = load i32, i32* %3, align 4, !dbg !16
  ret i32 %4, !dbg !17
}
```

Can we (optionally) filter those from the output? It clutters the code.

-Johan

**Comment by JohanEngelen** [comment]
(note: `-g` should not be removed from the compiler command, otherwise line information is no longer output)

**Comment by RabsRincon** [comment]
Thanks for the request Johan. This is mostly trivial to do but I can't find a propper place to place the toggle for this! I can use some already existing filter but then it would not be obvious at all. My first idea was to do exactly that - I don't have a lot of free time lately to test it so I'll leave the changes here in case anyone wants to pick this up. Sorry I can't be of more help!

```diff
diff --git a/lib/llvm-ir.js b/lib/llvm-ir.js
index 96bc1ac2..987c654d 100644
--- a/lib/llvm-ir.js
+++ b/lib/llvm-ir.js
@@ -106,6 +106,7 @@ class LlvmIrParser {
 
         // Filters
         const commentOnly = /^\s*(;.*)$/;
+        const debugLine = /@llvm\.dbg\.declare/;
 
         irLines.forEach(line => {
             let source = null;
@@ -124,6 +125,10 @@ class LlvmIrParser {
                 return;
             }
 
+            if (filters.libraryCode && line.match(debugLine)) {
+                return;
+            }
+
             // Non-Meta IR line. Metadata is attached to it using "!dbg !123"
             match = line.match(this.debugReference);
             if (match) {
```

**Comment by JohanEngelen** [comment]
About the filters: to be honest, it is a little annoying that the assembly filter buttons also apply to the IR output (not very obvious).

**Comment by jeremy-rifkin** [comment]
#1314 is about this as well

**Comment by jeremy-rifkin** [comment]
This is closed by #5045 and #5078 🎉
